src/golang/lib/engine/aq_engine.go

```diff
@@ -147,6 +147,15 @@ func (eng *aqEngine) ScheduleWorkflow(
 	return nil
 }
 
+func (eng *aqEngine) deleteTemporaryArtifactContents(ctx context.Context, dag dag_utils.WorkflowDag) {
```

Comment:
Can you note here that this is best effort. Also, that it only deletes the latest artifact result right?

Reply:
I think it's a bit weird to explicitly note about the latest result. Similar to the other commend, current dag / op / artf objects have a result included and is used to handle the lifecycle of 'one workflow execution'. And all operations over these objects are tied to that execution. In this way I feel which result to delete is clear from the context? Also technically, if we reconstruct a workflow from an older run, we can use the same method to delete contents from that run as well

Reply:
> current dag / op / artf objects have a result included and is used to handle the lifecycle of 'one workflow execution'.

I think I'm confused cus you can execute the same dag / op/ artf multiple times. An artifact is different than an artifact result so as soon as I see this method signature I'm thinking "Which artifact result am I deleting". But ok I think I see what you mean, an artf in orchestration means something different than an artifact in our data model? This seemed like implicit context I wanted to call out.

Reply:
I don't think our current abstraction allows us to re-execute a dag by calling `ExecuteWorkflow(dag)` multiple times? If you look at the constructor, we explicitly require a result ID to pass in to construct the `WorkflowDag` object: `NewWorkflowDag(..., dagResultID,...)` . In this way I think our abstraction is explicitly designed to have one object track one result.

I think it's a good idea to clarify this context somewhere, and I will add some comments to `WorkflowDag` object's definition. I just don't feel it's specific to this feature so we don't have to call out that on the specific API we added

Reply:
Yeah I'm not saying I want us to change the way we execute, I just wanted that context clarified somewhere cus I do think it's an important callout. Agreed we can put it at a higher layer (maybe in orchestration or as you said, in workflowDag)

Reply:
Sg! I updated comments for wfDag / artf / op objects, lmk if you have suggestions on where else we can clarify this